pkg/elemental/elemental.go

```diff
@@ -338,9 +341,11 @@ func (e *Elemental) DeployImage(img *v1.Image, leaveMounted bool) (info interfac
 		return nil, err
 	}
 	if !img.Source.IsFile() {
-		err = utils.CreateDirStructure(e.config.Fs, target)
-		if err != nil {
-			return nil, err
+		if createDirStructure {
```

Comment:
Let's avoid [the evil boolean arguments](https://medium.com/@MaLiN2223/my-boolean-vendetta-a2b473a64830) and create a new `DeployImageWithDir` method instead. It will run this new block in the beginning and then it will simply call `DeployImage`. What do you think?

Reply:
We still have deployImage accepting boolean arguments :D. With the methods you currently have, what I meant was to move this block that runs `CreateDirStructure` inside `DeployImage` itself and skip it completely from here and from `DeployImageNoDirs`. This way, only the method that want's dirs will call it and no (new) boolean arguments are needed.
Currently it runs in the middle of the `deployImage` function but it seems independent from the code above it so it could just run before the `deployImage` call. E.g.

```
func (e *Elemental) DeployImage(img *v1.Image, leaveMounted bool) (info interface{}, err error) {
  if !img.Source.IsFile() {
    err = utils.CreateDirStructure(e.config.Fs, target)
    if err != nil {
      return nil, err
    }
  }
  return e.deployImage(img, leaveMounted)
}
```

and 

```
func (e *Elemental) DeployImageNodirs(img *v1.Image, leaveMounted bool) (info interface{}, err error) {
	return e.deployImage(img, leaveMounted)
}
```

Reply:
we could even get rid of `leaveMounted` in a similar way but that would produce 4 different methods (for all combinations). Let's skip it for now.

Reply:
discussed in Slack, it's more complex than this. Skipping for now.